From dfa26eda0ddba351687d2e59ebb9171c59d077fd Mon Sep 17 00:00:00 2001
From: David Pacheco <dap@joyent.com>
Date: Tue, 30 Aug 2016 13:27:50 -0700
Subject: [PATCH] MANTA-2974 manta-factoryreset crashes when no "manta"
 application is present Reviewed by: Jordan Paige Hendricks
 <jordan.hendricks@joyent.com> Approved by: Jordan Paige Hendricks
 <jordan.hendricks@joyent.com>

---
 cmd/manta-factoryreset.js | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/cmd/manta-factoryreset.js b/cmd/manta-factoryreset.js
index b9aa480..1ef99e2 100755
--- a/cmd/manta-factoryreset.js
+++ b/cmd/manta-factoryreset.js
@@ -192,6 +192,10 @@ async.waterfall([
 		var log = self.log;
 		log.info('checking stage');
 
+		if (!self.application) {
+			return (cb(null));
+		}
+
 		var app = self.application;
 
 		if (app.metadata.SIZE === 'production') {
-- 
2.21.0

